docs/contracts/hash-consensus.md

```diff
@@ -16,3 +16,428 @@ the slot returned from `IReportAsyncProcessor.getLastProcessingRefSlot`.
+
+```solidity
+function getReportProcessor() external view returns (address)
+```
+
+### getConsensusState()
+Returns info about the current frame and consensus state in that frame.
+
+```solidity
+function getConsensusState() external view returns (
+    uint256 refSlot,
+    bytes32 consensusReport,
+    bool isReportProcessing
+)
+```
+
+#### Returns
+
+Returns info about the current frame and consensus state in that frame.
+
+| Name                 | Type      | Description                                                           |
+| -------------------- | --------- | --------------------------------------------------------------------- |
+| `refSlot`            | `uint256` | Reference slot of the current reporting frame.                        |
+| `consensusReport`    | `bytes32` | Consensus report for the current frame, if any. Zero bytes otherwise. |
+| `isReportProcessing` | `bool`    | If consensus report for the current frame is already being processed. Consensus can be changed before the processing starts. |
+
+### getReportVariants()
+
+Returns report variants and their support for the current reference slot.
+```solidity
+function getReportVariants() external view returns (
+    bytes32[] memory variants,
+    uint256[] memory support
+)
+```
+
+### getConsensusStateForMember()
+
+Returns the extended information related to an oracle committee member with the
+given address and the current consensus state. Provides all the information needed for
+an oracle daemon to decide if it needs to submit a report.
+
+```solidity
+function getConsensusStateForMember(address addr) external view returns (MemberConsensusState memory result)
+```
+
+#### Parameters
+
+| Name   | Type      | Description         |
+| ------ | --------- | ------------------- |
+| `addr` | `address` | The member address. |
+
+#### Returns
+
+Returns a new type `MemberConsensusState`
+```solidity
+struct MemberConsensusState {
+    /// @notice Current frame's reference slot.
+    uint256 currentFrameRefSlot;
+
+    /// @notice Consensus report for the current frame, if any. Zero bytes otherwise.
+    bytes32 currentFrameConsensusReport;
+
+    /// @notice Whether the provided address is a member of the oracle committee.
+    bool isMember;
+
+    /// @notice Whether the oracle committee member is in the fast lane members subset
+    /// of the current reporting frame. See `getIsFastLaneMember`.
+    bool isFastLane;
+
+    /// @notice Whether the oracle committee member is allowed to submit a report at
+    /// the moment of the call.
+    bool canReport;
+
+    /// @notice The last reference slot for which the member submitted a report.
+    uint256 lastMemberReportRefSlot;
+
+    /// @notice The hash reported by the member for the current frame, if any.
+    /// Zero bytes otherwise.
+    bytes32 currentFrameMemberReport;
+}
+```
+
+
+## Methods
+
+### updateInitialEpoch()
+
+Sets a new initial epoch given that the current initial epoch is in the future.
```

Comment:
May explain further (we stumbled upon this during the Lido V2 upgrade :grinning: )